File: website-connection.js
// Website connection module for SavosBot Club
const axios = require('axios');

class WebsiteConnection {
  constructor() {
    this.websiteUrl = process.env.WEBSITE_URL || 'https://your-app.vercel.app';
    this.apiKey = process.env.API_KEY || 'savosbot2024';
  }

  // Send user data to website
  async sendUserToWebsite(userData) {
    try {
      const response = await axios.post(`${this.websiteUrl}/api/users`, {
        ...userData,
        source: 'telegram_bot'
      }, {
        headers: {
          'Authorization': `Bearer ${this.apiKey}`,
          'Content-Type': 'application/json'
        },
        timeout: 5000
      });

      console.log('✅ User data sent to website:', userData.id);
      return response.data;
    } catch (error) {
      console.error('❌ Error sending user to website:', error.message);
      return null;
    }
  }

  // Get user data from website
  async getUserFromWebsite(userId) {
    try {
      const response = await axios.get(`${this.websiteUrl}/api/users/${userId}`, {
        headers: {
          'Authorization': `Bearer ${this.apiKey}`,
          'Content-Type': 'application/json'
        },
        timeout: 5000
      });

      return response.data;
    } catch (error) {
      console.error('❌ Error getting user from website:', error.message);
      return null;
    }
  }

  // Update user data on website
  async updateUserOnWebsite(userId, updateData) {
    try {
      const response = await axios.put(`${this.websiteUrl}/api/users/${userId}`, {
        ...updateData,
        source: 'telegram_bot'
      }, {
        headers: {
          'Authorization': `Bearer ${this.apiKey}`,
          'Content-Type': 'application/json'
        },
        timeout: 5000
      });

      console.log('✅ User data updated on website:', userId);
      return response.data;
    } catch (error) {
      console.error('❌ Error updating user on website:', error.message);
      return null;
    }
  }

  // Send statistics to website
  async sendStatsToWebsite(stats) {
    try {
      const response = await axios.post(`${this.websiteUrl}/api/statistics`, {
        ...stats,
        source: 'telegram_bot',
        timestamp: new Date().toISOString()
      }, {
        headers: {
          'Authorization': `Bearer ${this.apiKey}`,
          'Content-Type': 'application/json'
        },
        timeout: 5000
      });

      console.log('✅ Statistics sent to website');
      return response.data;
    } catch (error) {
      console.error('❌ Error sending stats to website:', error.message);
      return null;
    }
  }

  // Check website health
  async checkWebsiteHealth() {
    try {
      const response = await axios.get(`${this.websiteUrl}/api/health`, {
        timeout: 5000
      });

      return response.data;
    } catch (error) {
      console.error('❌ Website health check failed:', error.message);
      return { status: 'error', message: error.message };
    }
  }

  // Send notification to website
  async sendNotificationToWebsite(notification) {
    try {
      const response = await axios.post(`${this.websiteUrl}/api/notifications`, {
        ...notification,
        source: 'telegram_bot',
        timestamp: new Date().toISOString()
      }, {
        headers: {
          'Authorization': `Bearer ${this.apiKey}`,
          'Content-Type': 'application/json'
        },
        timeout: 5000
      });

      console.log('✅ Notification sent to website');
      return response.data;
    } catch (error) {
      console.error('❌ Error sending notification to website:', error.message);
      return null;
    }
  }

  // Get website settings
  async getWebsiteSettings() {
    try {
      const response = await axios.get(`${this.websiteUrl}/api/settings`, {
        headers: {
          'Authorization': `Bearer ${this.apiKey}`,
          'Content-Type': 'application/json'
        },
        timeout: 5000
      });

      return response.data;
    } catch (error) {
      console.error('❌ Error getting website settings:', error.message);
      return null;
    }
  }

  // Update website settings
  async updateWebsiteSettings(settings) {
    try {
      const response = await axios.put(`${this.websiteUrl}/api/settings`, {
        ...settings,
        source: 'telegram_bot',
        updated_at: new Date().toISOString()
      }, {
        headers: {
          'Authorization': `Bearer ${this.apiKey}`,
          'Content-Type': 'application/json'
        },
        timeout: 5000
      });

      console.log('✅ Website settings updated');
      return response.data;
    } catch (error) {
      console.error('❌ Error updating website settings:', error.message);
      return null;
    }
  }
}

// Create singleton instance
const websiteConnection = new WebsiteConnection();

// Export functions
module.exports = {
  websiteConnection,
  WebsiteConnection
};

File: website_connection.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Website connection module for SavosBot Club
"""

import os
import json
import logging
from typing import Dict, List, Optional, Any
from datetime import datetime

import aiohttp

logger = logging.getLogger(__name__)

class WebsiteConnection:
    def __init__(self, website_url: str, api_key: str):
        self.website_url = website_url.rstrip('/')
        self.api_key = api_key
        self.timeout = aiohttp.ClientTimeout(total=10)
        
    async def send_user_to_website(self, user_data: Dict[str, Any]) -> Optional[Dict[str, Any]]:
        """Send user data to website"""
        try:
            payload = {
                **user_data,
                'source': 'telegram_bot'
            }
            
            async with aiohttp.ClientSession(timeout=self.timeout) as session:
                async with session.post(
                    f"{self.website_url}/api/users",
                    json=payload,
                    headers={
                        'Authorization': f'Bearer {self.api_key}',
                        'Content-Type': 'application/json'
                    }
                ) as response:
                    if response.status == 200:
                        result = await response.json()
                        logger.info(f"✅ User data sent to website: {user_data['id']}")
                        return result
                    else:
                        logger.error(f"❌ Failed to send user to website: {response.status}")
                        return None
                        
        except Exception as e:
            logger.error(f"❌ Error sending user to website: {e}")
            return None
            
    async def get_user_from_website(self, user_id: int) -> Optional[Dict[str, Any]]:
        """Get user data from website"""
        try:
            async with aiohttp.ClientSession(timeout=self.timeout) as session:
                async with session.get(
                    f"{self.website_url}/api/users/{user_id}",
                    headers={
                        'Authorization': f'Bearer {self.api_key}',
                        'Content-Type': 'application/json'
                    }
                ) as response:
                    if response.status == 200:
                        return await response.json()
                    else:
                        logger.error(f"❌ Failed to get user from website: {response.status}")
                        return None
                        
        except Exception as e:
            logger.error(f"❌ Error getting user from website: {e}")
            return None
            
    async def update_user_on_website(self, user_id: int, update_data: Dict[str, Any]) -> Optional[Dict[str, Any]]:
        """Update user data on website"""
        try:
            payload = {
                **update_data,
                'source': 'telegram_bot'
            }
            
            async with aiohttp.ClientSession(timeout=self.timeout) as session:
                async with session.put(
                    f"{self.website_url}/api/users/{user_id}",
                    json=payload,
                    headers={
                        'Authorization': f'Bearer {self.api_key}',
                        'Content-Type': 'application/json'
                    }
                ) as response:
                    if response.status == 200:
                        result = await response.json()
                        logger.info(f"✅ User data updated on website: {user_id}")
                        return result
                    else:
                        logger.error(f"❌ Failed to update user on website: {response.status}")
                        return None
                        
        except Exception as e:
            logger.error(f"❌ Error updating user on website: {e}")
            return None
            
    async def send_stats_to_website(self, stats: Dict[str, Any]) -> Optional[Dict[str, Any]]:
        """Send statistics to website"""
        try:
            payload = {
                **stats,
                'source': 'telegram_bot',
                'timestamp': datetime.now().isoformat()
            }
            
            async with aiohttp.ClientSession(timeout=self.timeout) as session:
                async with session.post(
                    f"{self.website_url}/api/statistics",
                    json=payload,
                    headers={
                        'Authorization': f'Bearer {self.api_key}',
                        'Content-Type': 'application/json'
                    }
                ) as response:
                    if response.status == 200:
                        result = await response.json()
                        logger.info("✅ Statistics sent to website")
                        return result
                    else:
                        logger.error(f"❌ Failed to send stats to website: {response.status}")
                        return None
                        
        except Exception as e:
            logger.error(f"❌ Error sending stats to website: {e}")
            return None
            
    async def check_website_health(self) -> Dict[str, Any]:
        """Check website health"""
        try:
            async with aiohttp.ClientSession(timeout=self.timeout) as session:
                async with session.get(f"{self.website_url}/api/health") as response:
                    if response.status == 200:
                        return await response.json()
                    else:
                        return {'status': 'error', 'message': f'HTTP {response.status}'}
                        
        except Exception as e:
            logger.error(f"❌ Website health check failed: {e}")
            return {'status': 'error', 'message': str(e)}
            
    async def send_notification_to_website(self, notification: Dict[str, Any]) -> Optional[Dict[str, Any]]:
        """Send notification to website"""
        try:
            payload = {
                **notification,
                'source': 'telegram_bot',
                'timestamp': datetime.now().isoformat()
            }
            
            async with aiohttp.ClientSession(timeout=self.timeout) as session:
                async with session.post(
                    f"{self.website_url}/api/notifications",
                    json=payload,
                    headers={
                        'Authorization': f'Bearer {self.api_key}',
                        'Content-Type': 'application/json'
                    }
                ) as response:
                    if response.status == 200:
                        result = await response.json()
                        logger.info("✅ Notification sent to website")
                        return result
                    else:
                        logger.error(f"❌ Failed to send notification to website: {response.status}")
                        return None
                        
        except Exception as e:
            logger.error(f"❌ Error sending notification to website: {e}")
            return None
            
    async def get_website_settings(self) -> Optional[Dict[str, Any]]:
        """Get website settings"""
        try:
            async with aiohttp.ClientSession(timeout=self.timeout) as session:
                async with session.get(
                    f"{self.website_url}/api/settings",
                    headers={
                        'Authorization': f'Bearer {self.api_key}',
                        'Content-Type': 'application/json'
                    }
                ) as response:
                    if response.status == 200:
                        return await response.json()
                    else:
                        logger.error(f"❌ Failed to get website settings: {response.status}")
                        return None
                        
        except Exception as e:
            logger.error(f"❌ Error getting website settings: {e}")
            return None
            
    async def update_website_settings(self, settings: Dict[str, Any]) -> Optional[Dict[str, Any]]:
        """Update website settings"""
        try:
            payload = {
                **settings,
                'source': 'telegram_bot',
                'updated_at': datetime.now().isoformat()
            }
            
            async with aiohttp.ClientSession(timeout=self.timeout) as session:
                async with session.put(
                    f"{self.website_url}/api/settings",
                    json=payload,
                    headers={
                        'Authorization': f'Bearer {self.api_key}',
                        'Content-Type': 'application/json'
                    }
                ) as response:
                    if response.status == 200:
                        result = await response.json()
                        logger.info("✅ Website settings updated")
                        return result
                    else:
                        logger.error(f"❌ Failed to update website settings: {response.status}")
                        return None
                        
        except Exception as e:
            logger.error(f"❌ Error updating website settings: {e}")
            return None
            
    async def sync_data_with_website(self, data_type: str, data: Dict[str, Any]) -> bool:
        """Sync data with website"""
        try:
            if data_type == 'user':
                return await self.send_user_to_website(data) is not None
            elif data_type == 'stats':
                return await self.send_stats_to_website(data) is not None
            elif data_type == 'settings':
                return await self.update_website_settings(data) is not None
            else:
                logger.error(f"Unknown data type: {data_type}")
                return False
                
        except Exception as e:
            logger.error(f"Error syncing data with website: {e}")
            return False
